src/Gui/ApplicationPy.cpp

```diff
@@ -1301,36 +1306,60 @@ PyObject* Application::sGetCommandShortcut(PyObject * /*self*/, PyObject *args)
     }
 }
 
+PyObject* Application::sSetCommandShortcut(PyObject * /*self*/, PyObject *args)
+{
+    char* pName;
+    char* pShortcut;
+    if (!PyArg_ParseTuple(args, "ss", &pName, &pShortcut))
+        return NULL;
+
+    Command* cmd = Application::Instance->commandManager().getCommandByName(pName);
+    if (cmd && cmd->getAction()) {
+        Action* action = cmd->getAction();
+        QKeySequence shortcut = QString::fromLatin1(pShortcut);
+        QString nativeText = shortcut.toString(QKeySequence::NativeText);
+        action->setShortcut(nativeText);
+        ParameterGrp::handle hGrp = WindowParameter::getDefaultParameter()->GetGroup("Shortcut");
+        hGrp->SetASCII(pName, pShortcut);
+        return Py::new_reference_to(Py::Boolean(true));
+    }
+    else {
+        PyErr_Format(Base::BaseExceptionFreeCADError, "No such command '%s'", pName);
+        return NULL;
+    }
+}
+
+
 PyObject* Application::sGetCommandInfo(PyObject * /*self*/, PyObject *args)
 {
     char* pName;
     if (!PyArg_ParseTuple(args, "s", &pName))
         return NULL;
-
     Command* cmd = Application::Instance->commandManager().getCommandByName(pName);
-    if (cmd) {
+    if (cmd && cmd->getAction()) {
+        Action* action = cmd->getAction();
         PyObject* pyList = PyList_New(6);
         const char* menuTxt = cmd->getMenuText();
         const char* tooltipTxt = cmd->getToolTipText();
         const char* whatsThisTxt = cmd->getWhatsThis();
         const char* statustipTxt = cmd->getStatusTip();
         const char* pixMapTxt = cmd->getPixmap();
-        const char* shortcutTxt = cmd->getAccel();
+        std::string shortcutTxt = action->shortcut().toString().toStdString();
 
 #if PY_MAJOR_VERSION >= 3
         PyObject* strMenuTxt = PyUnicode_FromString(menuTxt ? menuTxt : "");
         PyObject* strTooltipTxt = PyUnicode_FromString(tooltipTxt ? tooltipTxt : "");
         PyObject* strWhatsThisTxt = PyUnicode_FromString(whatsThisTxt ? whatsThisTxt : "");
         PyObject* strStatustipTxt = PyUnicode_FromString(statustipTxt ? statustipTxt : "");
         PyObject* strPixMapTxt = PyUnicode_FromString(pixMapTxt ? pixMapTxt : "");
-        PyObject* strShortcutTxt = PyUnicode_FromString(shortcutTxt ? shortcutTxt : "");
+        PyObject* strShortcutTxt = PyUnicode_FromString(!shortcutTxt.empty() ? shortcutTxt.c_str() : "")

```

Comment:
It seems the final semi-colon `;` is missing here. The unit test with Python 3 fails because of it.